core/module.h

```diff
@@ -192,14 +183,44 @@ class ModuleBuilder {
     return ret;
   }
 
-  static std::string GenerateDefaultName(const std::string &class_name,
-                                         const std::string &default_template);
-
   CommandResponse RunCommand(Module *m, const std::string &user_cmd,
                              const google::protobuf::Any &arg) const;
 
   CommandResponse RunInit(Module *m, const google::protobuf::Any &arg) const;
 
+ private:
+  const std::function<Module *()> module_generator_;
+
+  const gate_idx_t num_igates_;
+  const gate_idx_t num_ogates_;
+
+  const std::string class_name_;
+  const std::string name_template_;
+  const std::string help_text_;
+  const Commands cmds_;
+  const module_init_func_t init_func_;
+};
+
+// A static class for managing module graphs
+class ModuleGraph {
+ public:
+
+  // Return true if any module from the builder exists
+  static bool ExistModuleClass(const ModuleBuilder *);
```

Comment:
nit: `HasModuleOfClass` might be a better name (although I'm not sure I really like this name either—the tricky bit is that we're asking whether any module exists using the class _implied by_ the given `ModuleBuilder` instance, which is hard to encode in the predicate name...).

Reply:
Really, it is better!